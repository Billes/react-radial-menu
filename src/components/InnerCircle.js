import React from 'react'
import cxs from 'cxs/monolithic'
import {getX, getY} from '../utils/calculations'

const s = {
  innerCircle: {
    fill: '#fff',

    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    color: '#404040'
  },
  innerContent: {
    overflow: 'visible'
  },
  text: {
    letterSpacing: '1px',
    fontWeight: 400,
    fill: '#404040'
  }
}

const circlePercentage = 0.57

const InnerCircle = ({current, diameter}) => {
  const outerRadius = diameter / 2
  const innerRadius = outerRadius * circlePercentage
  let title = current.name ||  'Menu'
  let image = current.image || null
  let innerAdjacent = Math.sqrt(2) * innerRadius

  return (
    <g>
      <circle className={cxs(s.innerCircle)} cx={outerRadius} cy={outerRadius} r={innerRadius} />
      <svg className={cxs(s.innerContent)} x={getX(outerRadius,innerRadius, -135)} y={getY(outerRadius,innerRadius, -135)} width={innerAdjacent} height={innerAdjacent} >
        <image
          style={{color: '#404040'}}
          width={`${innerAdjacent * 0.7}px`}
          height={`${innerAdjacent * 0.7}px`}
          x={(innerAdjacent - (innerAdjacent * 0.7)) / 2} y={innerAdjacent * 0.05}
          xlinkHref={image}
        />
        <text x={innerAdjacent/2} y={innerAdjacent * 0.95} className={cxs(s.text)}>
          <tspan textAnchor="middle">{title}</tspan>
        </text>
      </svg>
    </g>
  )
}

InnerCircle.propTypes = {
  current: React.PropTypes.object,
  diameter: React.PropTypes.number,
}

export default InnerCircle
