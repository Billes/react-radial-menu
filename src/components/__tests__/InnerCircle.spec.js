/*global expect, test */
import React from 'react'
import renderer from 'react-test-renderer'
import InnerCircle from '../InnerCircle'

const component = (props) => renderer.create(
  <InnerCircle {...props} />
)

test('render when current is set', () => {
  let props = {
    item: {
      action: 'http://google.se',
      image: '/path/to/img',
      name: 'whoops'
    },
    outerRadius: 200
  }

  let tree = component(props).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render when current is not set', () => {
  let props = {
    outerRadius: 200
  }

  let tree = component(props).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render when no image is set', () => {
  let props = {
    item: {
      action: 'http://google.se',
      name: 'whoops'
    },
    outerRadius: 200
  }

  let tree = component(props).toJSON()
  expect(tree).toMatchSnapshot()
})
