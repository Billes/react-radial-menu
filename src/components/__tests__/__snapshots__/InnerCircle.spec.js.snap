// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when current is not set 1`] = `
<g>
  <circle
    cx={200}
    cy={200}
    r={113.99999999999999}
    style={
      Object {
        "alignItems": "center",
        "color": "#404040",
        "display": "flex",
        "fill": "#fff",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  />
  <svg
    height={161.22034611053283}
    width={161.22034611053283}
    x={119.3898269447336}
    y={119.38982694473358}
  >
    <text
      style={
        Object {
          "fill": "#404040",
          "fontWeight": 400,
          "letterSpacing": "1px",
        }
      }
      x={80.61017305526642}
      y={153.15932880500617}
    >
      <tspan
        textAnchor="middle"
      >
        Menu
      </tspan>
    </text>
  </svg>
</g>
`;

exports[`render when current is set 1`] = `
<g>
  <circle
    cx={200}
    cy={200}
    r={113.99999999999999}
    style={
      Object {
        "alignItems": "center",
        "color": "#404040",
        "display": "flex",
        "fill": "#fff",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  />
  <svg
    height={161.22034611053283}
    width={161.22034611053283}
    x={119.3898269447336}
    y={119.38982694473358}
  >
    <image
      height="112.85424227737298px"
      style={
        Object {
          "color": "#404040",
        }
      }
      width="112.85424227737298px"
      x={24.183051916579927}
      xlinkHref="/path/to/img"
      y={8.061017305526642}
    />
    <text
      style={
        Object {
          "fill": "#404040",
          "fontWeight": 400,
          "letterSpacing": "1px",
        }
      }
      x={80.61017305526642}
      y={153.15932880500617}
    >
      <tspan
        textAnchor="middle"
      >
        whoops
      </tspan>
    </text>
  </svg>
</g>
`;

exports[`render when no image is set 1`] = `
<g>
  <circle
    cx={200}
    cy={200}
    r={113.99999999999999}
    style={
      Object {
        "alignItems": "center",
        "color": "#404040",
        "display": "flex",
        "fill": "#fff",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  />
  <svg
    height={161.22034611053283}
    width={161.22034611053283}
    x={119.3898269447336}
    y={119.38982694473358}
  >
    <text
      style={
        Object {
          "fill": "#404040",
          "fontWeight": 400,
          "letterSpacing": "1px",
        }
      }
      x={80.61017305526642}
      y={153.15932880500617}
    >
      <tspan
        textAnchor="middle"
      >
        whoops
      </tspan>
    </text>
  </svg>
</g>
`;
